ocs_ci/ocs/node.py

```diff
@@ -908,6 +914,91 @@ def delete_and_create_osd_node_vsphere_upi(osd_node_name, use_existing_node=Fals
+    """
+    Unschedule, drain and delete osd node, and creating a new osd node.
+    At the end of the function there should be the same number of osd nodes as
+    it was in the beginning, and also ceph health should be OK.
+    This function is for vSphere UPI.
+
+    Args:
+        osd_node_name (str): the name of the osd node
+        use_existing_node (bool): If False, create a new node and label it.
+            If True, use an existing node to replace the deleted node
+            and label it.
+
+    Returns:
+        str: The new node name
+
+    """
+    from ocs_ci.ocs.platform_nodes import PlatformNodesFactory
+    from ocs_ci.ocs.resources.storage_cluster import get_osd_size
+
+    sc_name = constants.LOCAL_BLOCK_RESOURCE
+    old_pv_objs = get_pv_objs_in_sc(sc_name)
+
+    osd_node = get_node_objs(node_names=[osd_node_name])[0]
+    osd_pod = get_node_pods(osd_node_name, pods_to_search=pod.get_osd_pods())[0]
+    osd_id = pod.get_osd_pod_id(osd_pod)
+    log.info(f"osd id to remove = {osd_id}")
+
+    log.info("Scale down node deployments...")
+    scale_down_deployments(osd_node_name)
+    log.info("Scale down deployments finished successfully")
+
+    new_node_name = delete_and_create_osd_node_vsphere_upi(
+        osd_node_name, use_existing_node
+    )
+    assert new_node_name, "Failed to create a new node"
+    log.info(f"New node created successfully. Node name: {new_node_name}")
+
+    # If we use LSO, we need to create and attach a new disk manually
+    new_node = get_node_objs(node_names=[new_node_name])[0]
+    plt = PlatformNodesFactory()
+    node_util = plt.get_nodes_platform()
+    osd_size = get_osd_size()
+    log.info(
+        f"Create a new disk with size {osd_size}, and attach to node {new_node_name}"
+    )
+    node_util.create_and_attach_volume(node=new_node, size=osd_size)
+
+    log.info(
+        "Replace the old node with the new worker node in localVolumeDiscovery and localVolumeSet"
+    )
+    res = add_new_node_to_lvd_and_lvs(
+        old_node_name=get_node_hostname_label(osd_node),
+        new_node_name=get_node_hostname_label(new_node),
+    )
+    assert res, "Failed to add the new node to LVD and LVS"
+
+    log.info("Verify new pv is available...")
+    is_new_pv_available = verify_new_pv_available_in_sc(old_pv_objs, sc_name)
+    assert is_new_pv_available, "New pv is not available"
+    log.info("Finished verifying that the new pv is available")
+
+    switch_to_project(defaults.ROOK_CLUSTER_NAMESPACE)
+
+    osd_removal_job = pod.run_osd_removal_job(osd_id)
+    assert osd_removal_job, "ocs-osd-removal failed to create"
+    is_completed = (pod.verify_osd_removal_job_completed_successfully(osd_id),)
+    assert is_completed, "ocs-osd-removal-job is not in status 'completed'"
+    log.info("ocs-osd-removal-job completed successfully")
+
+    delete_released_pvs_in_sc(sc_name)
+
+    # Delete the crashcollector pod deployment
```

Comment:
Does this step is required? I see above we are scaling down the crashcollector pod deployment as well

Reply:
Yes, this is true. But from the docs, I see that we need to perform this step too.

Reply:
i guess we dont need to delete crashcollector pods